libs/libcommon/src/libcommon/parquet_utils.py

```diff
@@ -139,9 +157,24 @@ def query(self, offset: int, length: int) -> pa.Table:
                 row_group_offsets, [first_row, last_row], side="right"
             )
 
+        with StepProfiler(
+            method="parquet_index_with_metadata.row_groups_size_check", step="check if the rows can fit in memory"
+        ):
+            row_groups_size = sum(
+                [row_group_readers[i].read_size() for i in range(first_row_group_id, last_row_group_id + 1)]
+            )
+            if row_groups_size > self.max_arrow_data_in_memory:
+                raise TooBigRows(
```

Comment:
maybe mention in the `query` function docstring that it can raise this exception, and when